test/src/Provider/AbstractProviderTest.php

```diff
@@ -212,6 +215,60 @@ public function testErrorResponsesCanBeCustomizedAtTheProvider()
         $this->assertEquals('Foo error', $errorMessage);
         $this->assertEquals(1337, $errorCode);
     }
+
+    public function testClientErrorTriggersHttpException()
+    {
+        $provider = new MockProvider([
+          'clientId' => 'mock_client_id',
+          'clientSecret' => 'mock_secret',
+          'redirectUri' => 'none',
+        ]);
+
+        $response = m::mock('Ivory\HttpAdapter\Message\ResponseInterface');
+        $response->shouldReceive('getBody')
+                 ->times(1)
+                 ->andReturn('{"error":"BadResponse","code":500}');
+
+        $exception = new HttpAdapterException();
```

Comment:
This test looks good. Frankly, I'm not sure why the code coverage was being ignored all over the place. That's been on my list of things to clean up for v1.0.